beacon_node/beacon_chain/src/light_client_server_cache.rs

```diff
@@ -419,46 +434,42 @@ impl<T: BeaconChainTypes> LightClientServerCache<T> {
         finalized_period: u64,
         chain_spec: &ChainSpec,
     ) -> Result<Option<(LightClientBootstrap<T::EthSpec>, ForkName)>, BeaconChainError> {
-        if let Some(block) = store.get_full_block(block_root)? {
-            let (state_root, slot) = (block.state_root(), block.slot());
-
-            let fork_name = chain_spec.fork_name_at_slot::<T::EthSpec>(slot);
+        let Some(block) = store.get_full_block(block_root)? else {
+            return Ok(None);
+        };
+        
+        let (_, slot) = (block.state_root(), block.slot());
 
-            let sync_committee_period = block
-                .slot()
-                .epoch(T::EthSpec::slots_per_epoch())
-                .sync_committee_period(chain_spec)?;
+        let fork_name = chain_spec.fork_name_at_slot::<T::EthSpec>(slot);
 
-            let current_sync_committee_branch =
-                self.get_sync_committee_branch(store, block_root)?.unwrap();
-            // TODO unwrap
-            // db.current_sync_committee_branch.get(block_root)
-
-            let current_sync_committee = if sync_committee_period <= finalized_period {
-                // fetch the header current sync committee from the db, since its not cached
-                // TODO unwrap
-                Arc::new(
-                    self.get_sync_committee(store, sync_committee_period)?
-                        .unwrap(),
-                )
-            } else {
-                // were at the current finalized period, so this data should be easily retrievable from the currently cached state
-                // TODO unwrap
-                let state = store.get_state(&state_root, Some(slot))?.unwrap();
-                state.current_sync_committee()?.clone()
-            };
+        let sync_committee_period = block
+            .slot()
+            .epoch(T::EthSpec::slots_per_epoch())
+            .sync_committee_period(chain_spec)?;
 
-            let light_client_bootstrap = LightClientBootstrap::new(
-                &block,
-                current_sync_committee,
-                current_sync_committee_branch,
-                chain_spec,
-            )?;
+        let Some(current_sync_committee_branch) =
+            self.get_sync_committee_branch(store, block_root)?
+        else {
+            return Ok(None);
+        };
 
-            return Ok(Some((light_client_bootstrap, fork_name)));
+        if sync_committee_period > finalized_period {
+            return Ok(None);
         }
 
-        Ok(None)
+        let Some(current_sync_committee) = self.get_sync_committee(store, sync_committee_period)?
```

Comment:
In this function we read three pieces of data: 
- `get_full_block`
- `get_sync_committee_branch`
- `get_sync_committee`

If we have the block but not the branch, what does that mean? Someone may request a block root that does not exist but I feel we should be stricter about our DB consistency. Can we assure that if a block exist, there must exist its `sync_committee_branch` and `sync_committee`?

Reply:
it should be the case that a `sync_committee_branch` and `sync_committee` exist for a finalized block root as long as that block root was processed by the node after initial sync.

so if start my node up, it finishes syncing, and immediately after syncing my head is at block 100, `sync_committee_branch` and `sync_committee` will begin persisting for blocks after block 100. if we have the block but not the branch + sync committee, its probably because that block was persisted during the nodes initial sync process OR it was persisted before the node enabled light client functionality. 

I think we'd need a backfill mechanism to ensure that a finalized block root always has an associated `sync_committee_branch` and `sync_committee`

Reply:
also if the provided block root isnt at an epoch boundary, we wouldnt have the `sync_committee_branch` for it

Reply:
Can you add comments in the code reasoning about each case, with examples of when things may be missing? Also for better debuggability we should not return the same `None` for everything. The API route [`getLightClientBootstrap`](https://github.com/ethereum/beacon-APIs/blob/515216384db1f17f2b5f3ab8a0fc39340db8ee9d/apis/beacon/light_client/bootstrap.yaml#L55) supports a message, so we can differentiate 
- 404: block unknown
- 404: no sync committee branch
- 404: no sync committee

Same for the p2p route, we can attach messages on a resource unavailable response

Reply:
added better comments, and returned helpful error messages here: 
e75b84f